src/Agent.Listener/Capabilities/AgentCapabilitiesProvider.cs

```diff
@@ -34,6 +35,10 @@ public Task<List<Capability>> GetCapabilitiesAsync(AgentSettings settings, Cance
 #if OS_WINDOWS
             Add(capabilities, "Agent.OSVersion", GetOSVersionString());
             Add(capabilities, "Cmd", Environment.GetEnvironmentVariable("comspec"));
+
+            //This capability gets set irrespective of agent being run in auto-logon mode or not.
+            var configurationManager = HostContext.GetService<IConfigurationManager>();
+            Add(capabilities, "InteractiveSession", (!configurationManager.IsServiceConfigured()).ToString());
```

Comment:
I firmly do not approve of the decision to won't fix this. This should either be fixed or removed.

The correct solution does not require much additional engineering. I'm not OK with the proposed "it's accurate sometimes" approach.